lib/cli/interactive-setup/console-instrument.js

```diff
@@ -0,0 +1,202 @@
+'use strict';
+const wait = require('timers-ext/promise/sleep');
+const { log, progress } = require('@serverless/utils/log');
+const resolveAuthMode = require('@serverless/utils/auth/resolve-mode');
+const apiRequest = require('@serverless/utils/api-request');
+
+const successMessage = 'Your functions are instrumented for development.';
+const progressKey = 'instrumentation-progress';
+
+const allFunctionsExist = async (context) => {
+  const { total } = await apiRequest(`/api/search/orgs/${context.org.orgId}/search`, {
+    method: 'POST',
+    body: {
+      from: 0,
+      size: context.targetFunctions.length,
+      query: {
+        bool: {
+          must: [
+            {
+              match: { type: 'resource_aws_lambda' },
+            },
+            {
+              match: { tag_account_id: context.awsAccountId },
+            },
+            {
+              terms: { 'aws_lambda_name.keyword': context.targetFunctions },
+            },
+          ],
+        },
+      },
+    },
+  });
+
+  return {
+    allExist: total === context.targetFunctions.length,
+    total: context.targetFunctions.length,
+    functionCount: total,
+  };
+};
+
+const checkInstrumentationStatus = async (context) => {
+  const { total } = await apiRequest(`/api/search/orgs/${context.org.orgId}/search`, {
+    method: 'POST',
+    body: {
+      from: 0,
+      size: context.targetFunctions.length,
+      query: {
+        bool: {
+          must: [
+            {
+              match: { type: 'resource_aws_lambda' },
+            },
+            {
+              match: { tag_account_id: context.awsAccountId },
+            },
+            {
+              match: { instrument_mode: 'dev' },
+            },
+            {
+              terms: { 'aws_lambda_name.keyword': context.targetFunctions },
+            },
+          ],
+        },
+      },
+    },
+  });
+
+  return {
+    isInstrumented: total === context.targetFunctions.length,
+    total: context.targetFunctions.length,
+    instrumented: total,
+  };
+};
+
+const waitForInstrumentation = async (context) => {
+  const instrumentationProgress = progress.get(progressKey);
+  let isInstrumenting = true;
+  while (isInstrumenting) {
+    const { isInstrumented: done, total, instrumented } = await checkInstrumentationStatus(context);
+    instrumentationProgress.update(`Instrumenting ${instrumented}/${total} functions`);
+    if (done) {
+      isInstrumenting = false;
+    } else {
+      await wait(1000);
+    }
+  }
+};
+
+module.exports = {
+  async isApplicable(context) {
+    const { isConsole, enableDev } = context;
+
+    if (!isConsole) {
+      context.inapplicabilityReasonCode = 'NON_CONSOLE_CONTEXT';
+      return false;
+    }
+
+    if (!enableDev) {
```

Comment:
Since we are removing the onboarding steps there is no need to prompt.

Reply:
> Since we are removing the onboarding steps 

What steps are we removing?

Reply:
Sorry I should have specified, removing the **_dev mode_** onboarding steps 😅

Reply:
This PR is about adding dev mode onboarding steps. So you mean it's not longer applicable?

Reply:
The task I was assigned in linear was originally just supposed to be activating the dev mode feed on an existing serverless project.

There really wasn't a requirement for us to add the enable dev mode or dev mode feed to the CLI onboarding steps. I just thought it would be simple enough to add dev mode to the CLI "onboarding" and existing projects since they are the same general steps.

But if we can't move the `deploy` step around in the onboarding flow then we can't allow people to enable dev mode when creating a new project since the deploy happens after the integration is created and the delay can cause the dev mode instrumentation to fail.

Reply:
In my understanding, what this task needs is in 80% covered by onboarding, and it simply feels as an onboarding extension.

> But if we can't move the deploy step around 

Still, it's not clear why there's a need for that. In the case of (1) existing service, `deploy` will not happen, and in case of (2) not existing service, we do not apply dev mode anyway. So to me, it feels like `deploy` is totally irrelevant.

Reply:
You are correct and I think we are saying the same thing here 😅 I reverted the steps in the index file so I think we should be good or did I need to move one of the new steps around?

Reply:
ok, I got lost a bit 😅